test/stablecoin_converter.js

```diff
@@ -0,0 +1,110 @@
+      assert.equal(((await stablecoinConverter.orders.call(user_1,id)).sellToken).toNumber(), 1, "sellToken was stored incorrectly")
+      assert.equal(((await stablecoinConverter.orders.call(user_1,id)).buyToken).toNumber(), 0, "buyToken was stored incorrectly")
+      assert.equal(((await stablecoinConverter.orders.call(user_1,id)).sellOrderFlag), true, "sellTokenFlag was stored incorrectly")
+      assert.equal(((await stablecoinConverter.orders.call(user_1,id)).validFrom).toNumber(), currentStateIndex.toNumber() + 1, "validFrom was stored incorrectly")
+      assert.equal(((await stablecoinConverter.orders.call(user_1,id)).validTill).toNumber(), 3, "validTill was stored incorrectly")
+    })
+  })
+  describe("cancelOrder", () => {
+    it("places orders, then cancels it and orders status", async () => {
+      const stablecoinConverter = await StablecoinConverter.new()
+
+      const id = await stablecoinConverter.placeOrder.call(0, 1, true, 3, 10, 20, {from: user_1})
+      await stablecoinConverter.placeOrder(0, 1, true, 3, 10, 20, {from: user_1})
+      const currentStateIndex = await stablecoinConverter.getCurrentStateIndex()
+      await stablecoinConverter.cancelOrder(id, {from: user_1})
+
+      assert.equal(((await stablecoinConverter.orders.call(user_1,id)).validTill).toNumber(), currentStateIndex, "validTill was stored incorrectly")
+    })
+  })
+  describe("deleteOrder", () => {
+    it("places a order, then cancels and deletes it", async () => {
+      const stablecoinConverter = await StablecoinConverter.new()
+
+      const id = await sendTxAndGetReturnValue(stablecoinConverter.placeOrder, 0, 1, true, 3, 10, 20)
+      await stablecoinConverter.cancelOrder(id)
+      await waitForNSeconds(300)
+      await stablecoinConverter.deleteOrder(id)
+
+      assert.equal((await stablecoinConverter.orders(user_1, id)).sellAmount, 0, "sellAmount was stored incorrectly")
+    })
+    it("fails to delete non-canceled order", async () => {
+      const stablecoinConverter = await StablecoinConverter.new()
+      const currentStateIndex = await stablecoinConverter.getCurrentStateIndex()
+
+      const id = await sendTxAndGetReturnValue(stablecoinConverter.placeOrder, 0, 1, true, currentStateIndex + 3, 10, 20)
+      await truffleAssert.reverts(stablecoinConverter.deleteOrder(id), "Order is still valid")
+    })
+    it("fails to delete canceled order in same stateIndex", async () => {
+      const stablecoinConverter = await StablecoinConverter.new()
+    
+      const id = await sendTxAndGetReturnValue(stablecoinConverter.placeOrder, 0, 1, true, 3, 10, 20)
+      await stablecoinConverter.cancelOrder(id)
+      await truffleAssert.reverts(stablecoinConverter.deleteOrder(id), "Order is still valid")
+    })
+  })
+  describe("addToken()", () => {
+    it("Anyone can add tokens", async () => {
+      const instance = await StablecoinConverter.new()
+
+      const token_1 = await ERC20.new()
```

Comment:
Not sure if this should be an instance of the MockContract instead. For now, doesn't seem to make a difference.

Reply:
Yes, this kindof introduces another dependency to the code base. We use `MintbleERC20`, `MockContract` and `ERC20` now all for the same things. Maybe it doesn't matter.